### DISCUSSION TITLE
Creating a Component with Variations

### DISCUSSION THREAD

**Discussion by thecodeclimber** [chakra-ui/chakra-ui#2010] [General] [answered]
I am trying to create some typography components using the `Heading` and `Text` components.  The idea is that these components contain within them all the line-height, letter-spacing, font-weight and other styles that I want in one component.

So, for example, a `<DisplayOne>` component has `font-size` of 96px, a `line-height` of 100%, etc.  A `<DisplayThree>` component has a `font-weight` and `line-height` of 72px.  And so.  

Now, this works, but it's not as elegant as I would like it.  Ideally, I'd like something like this:

```
<Heading level="d1">Display One</Heading>
<Heading level="d2">Display Two</Heading>
<Heading level="d3">Display Three</Display>
...
<Text level="small">Small Text</Text>
...
```
In other words, rather then have multiple components, have one component with a prop to control the variations.  I'm having trouble, though, figuring out how to get that to work.

Right now, this is what my code looks like:

```
import { Heading, Text } from '@chakra-ui/core'

const TextBase = ({ children, ...rest }) => (
  <Text color="darkGrey" fontWeight="semibold" lineHeight="base">
    {children}
  </Text>
)

const HeadingBase = ({ children, ...rest }) => (
  <Heading color="dark" fontWeight="bold" letterSpacing="tight">
    {children}
  </Heading>
)

// Display
export const DisplayOne = ({ children, ...rest }) => (
  <HeadingBase fontSize="96px" lineHeight="100%" {...rest}>
    {children}
  </HeadingBase>
)

export const DisplayTwo = ({ children, ...rest }) => (
  <HeadingBase fontSize="88px" lineHeight="100%" {...rest}>
    {children}
  </HeadingBase>
)

...

// Text
export const BodyText = ({ children, ...rest }) => (
  <TextBase fontSize="md" {...rest}>
    {children}
  </TextBase>
)

export const SmallText = ({ children, ...rest }) => (
  <TextBase fontSize="sm" {...rest}>
    {children}
  </TextBase>
)
...
```

What I thought to do, though, is something like this:

```
import { Heading as HeadingBase, Text as TextBase } from '@chakra-ui/core'```

export const Heading = (props) => {
  if (props.level === "d1") {
     return <Heading fontSize="96px"...>{props.children}</Heading>
} else if (props.level === "d2") { ... }
 ...
}
}
```

However, this seems rather involved, not very dry and I am wondering if there is a simpler and/or more elegant way to do this.

Any ideas?

Thanks.

**Comment by chasinhues** [comment, marked as answer]
@thecodeclimber sorry for the lack of responses on our part! Chakra may have not had this when you created this discussion, but check out [text styles](https://chakra-ui.com/docs/features/text-and-layer-styles#text-styles)
